samples/cloud-run/README.md

```diff
@@ -0,0 +1,131 @@
+# PGAdapter Cloud Run Sidecar
+
+This directory contains samples for deploying an application to Cloud Run with PGAdapter as a
+[sidecar container](https://cloud.google.com/run/docs/deploying#sidecars). This is the recommended
+setup on Cloud Run for PGAdapter. Take a look at the concrete examples in the subdirectories for a
+working example in a specific programming language:
+- [Java JDBC](./java)
+- [Go pgx](./golang)
+- [Node.js node-postgres](./nodejs)
+- [Python psycopg3](./python)
+- [.NET C# npgsql](./dotnet)
+- [Ruby ruby-pg](./ruby)
+
+The main application and PGAdapter communicate over a Unix domain socket using an
+[in-memory shared volume](https://cloud.google.com/run/docs/configuring/in-memory-volumes).
+This guarantees the lowest possible latency between your application and PGAdapter.
+
+The [`service.yaml`](./service.yaml) file in this directory contains an example setup for a Cloud
+Run application that uses PGAdapter as a sidecar proxy. The following sections explains the
+different parts of the configuration that is used to set up PGAdapter as a sidecar.
+
+## Sidecar Configuration
+
+PGAdapter is added to the service file as a separate container:
+
+```yaml
+spec:
+  template:
+    # ...
+    spec:
+      # ...
+      containers:
+        # This is the PGAdapter sidecar container.
+        - name: pgadapter
+          image: gcr.io/cloud-spanner-pg-adapter/pgadapter
+          volumeMounts:
+            - mountPath: /sockets
+              name: sockets-dir
+          args:
+            - -dir /sockets
+            - -x
+```
+
+This container specification adds the pre-built PGAdapter Docker image
+`gcr.io/cloud-spanner-pg-adapter/pgadapter` as a sidecar container to the setup. Cloud Run
+recognizes the container as a sidecar, because it does not specify a `containerPort`. Only the main
+container of a Cloud Run service may specify a `containerPort`.
+
+The configuration also adds a volume that is used for Unix domain sockets.
+See [Unix Domain Sockets](#unix-domain-sockets) for more information on this.
+
+PGAdapter is configured with two command line arguments in this example:
+1. `-dir /sockets`: This instructs PGAdapter to use the directory `/sockets` for setting up a
+   listener for incoming Unix Domain Socket connection requests.
+2. `-x`: This instructs PGAdapter to accept TCP connection requests from other hosts than localhost.
+   This is necessary for the TCP prober in the service file. _NOTE_: This will not make PGAdapter
+   accessible from the Internet. Cloud Run only allows ingress traffic to the main container.
```

Comment:
Slight rewording, and I'm probably being paranoid here, but may be reword to
"Cloud Run allows ingress traffic only to..."